Buildings/Controls/OBC/Utilities/PIDWithAutotuning/Relay/NormalizedTimeDelay.mo

```diff
@@ -0,0 +1,117 @@
+within Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay;
+block NormalizedTimeDelay
+  "Calculate the normalized time delay of a response from a relay controller"
+  parameter Real gamma(min=1+1E-6) = 4
+  "Asymmetry level of the relay controller";
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput rho
+    "Half period ratio"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+    iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput tau
+    "Normalized time delay"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.CDL.Continuous.AddParameter addPar1(
+    final p=-1)
+    "Calculate the difference between gamma and 1"
+    annotation (Placement(transformation(extent={{-40,-20},{-20,0}})));
+  Buildings.Controls.OBC.CDL.Continuous.AddParameter addPar2(
+    final p=0.65)
+    "Calculate the sum of the output of gai and 0.65"
+    annotation (Placement(transformation(extent={{-40,-70},{-20,-50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant asyLev(
+    final k=gamma)
+    "Asymmetry level of the relay controller"
+    annotation (Placement(transformation(extent={{-80,70},{-60,90}})));
+  Buildings.Controls.OBC.CDL.Continuous.Divide div "Calculate the normalized time delay"
+    annotation (Placement(transformation(extent={{40,-10},{60,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract subGamRho
+    "Calculate the difference between the asymmetry level of the relay controller and the half period ratio"
+    annotation (Placement(transformation(extent={{0,10},{20,30}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiplyByParameter gai(
+    final k=0.35)
+    "Gain for the half period ratio"
+    annotation (Placement(transformation(extent={{-80,-70},{-60,-50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Multiply mul
+    "Calculate the product of the output of addPar1 and that of addPar2"
+    annotation (Placement(transformation(extent={{0,-48},{20,-28}})));
+  Buildings.Controls.OBC.CDL.Utilities.Assert assMes(
+    final message="The asymmetry level should be larger than the half period ratio.")
```

Comment:
I assume this assertion can trigger during a simulation. Can you give some advice to the user for how to fix this situation rather than only stating what went wrong?

Reply:
The advice has been added to the <code>message</code>.